### DISCUSSION TITLE
task fail cannot found log in airflow dag

### DISCUSSION THREAD

**Discussion by Jakin-Liu** [apache/airflow#20584] [General]
### Apache Airflow version

2.0.1

### What happened

I have many DAGs executed in the same time period. Every day, several DAGs fail to execute. There are many tasks in this DAG, but one task fails to execute, and no relevant log information will be generated after the task fails. What should I do.

error info is:

*** Unable to read remote log from gs://xxxxxxxxxxxx/logs/abcd_dag/xxxxx/2021-12-28T17:00:00+00:00/5.log
*** 404 GET https://storage.googleapis.com/download/storage/v1/b/xxxxxxxx/o/logs%2Fxxxxxxx_dag%2Fxxxxx%2F2021-12-28T17%3A00%3A00%2B00%3A00%2F5.log?alt=media: No such object: xxxxxx/logs/xxxxxx_dag/xxxxxx_Redeem/2021-12-28T17:00:00+00:00/5.log: ('Request failed with status code', 404, 'Expected one of', <HTTPStatus.OK: 200>, <HTTPStatus.PARTIAL_CONTENT: 206>)

*** Log file does not exist: /home/airflow/gcs/logs/xxxxxx_dag/xxxxx/2021-12-28T17:00:00+00:00/5.log
*** Fetching from: http://:8793/log/xxxxx_dag/xxxxxxx_Redeem/2021-12-28T17:00:00+00:00/5.log
*** Failed to fetch log file from worker. Invalid URL 'http://:8793/log/xxxxxxxx/xxxx/2021-12-28T17:00:00+00:00/5.log': No host supplied

### Operating System

os

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by dstandish** [comment]
tell us a bit about your infra setup. are you using official helm chart?  celery? k8s executor? k8s pod operator?

When using blob storage like gcs, the logs are only uploaded after completion of the ttask.  If it is interrupted for some reason (e.g. OOM error on the worker) then the task will die immedietly and the logs will not be uploaded.  Depending on your infra there may be a solution for you.

**Comment by potiuk** [comment]
@dstandish  is tight - this is niot really reproducible case so there is no way we can qualify this as a bug. Until we have more information provided and some reproducibility, I turn that into discussion. Please provide more info on your case @Jakin-Liu
